tracing-core/src/span.rs

```diff
@@ -69,6 +69,7 @@ impl Id {
     }
 
     /// Returns the span's ID as a  `u64`.
+    #[allow(clippy::wrong_self_convention)]
     pub fn into_u64(&self) -> u64 {
```

Comment:
@hawkw It is possible to remove this allow by deriving `Copy` for `pub struct Id(NonZeroU64)` and passing the argument by value.

However, if I do this, it triggers several more warnings where clippy suggests basically always pass `Id` by value.

So, for instance, the method signatures for `Subscriber` would be changed from:

```rust
    fn record_follows_from(&self, span: &span::Id, follows: &span::Id);
```

to

```rust
    fn record_follows_from(&self, span: span::Id, follows: span::Id);
```

I started to do this actually but it's a pretty extensive change and requires a lot of updates.

There is one place where a new `#[allow(clippy::clone_of_copy)]` would be needed, which is the `Subscriber::clone_span` implementation for `Id` (since it sounds like that was being used for counting).

What do you think?

Reply:
> It is possible to remove this allow by deriving `Copy` for `pub struct Id(NonZeroU64)` and passing the argument by value.

I'd prefer not to implement `Copy` for `Id`. The type is trivially copyable _now_, but in order to hedge bets against future changes (such as adding more fields to `Id`), we chose not to implement `Copy` for it. If we were to add a field that was not trivially `Copy` in the future, we would have to remove the `Copy` impl, which would be a large breaking change.



> So, for instance, the method signatures for `Subscriber` would be changed from:
> 
> ```rust
>     fn record_follows_from(&self, span: &span::Id, follows: &span::Id);
> ```
> 
> to
> 
> ```rust
>     fn record_follows_from(&self, span: span::Id, follows: span::Id);
> ```
> 
> I started to do this actually but it's a pretty extensive change and requires a lot of updates.

Similarly, this is also a breaking API change that I would prefer not to make. I'd rather just add the `allow` attributes as needed, but with a comment explaining why the lint is allowed (if possible).